relayer/pkg/chainlink/ocr2/providers.go

```diff
@@ -77,14 +79,14 @@ type medianProvider struct {
 	reportCodec        median.ReportCodec
 }
 
-func NewMedianProvider(chainID string, contractAddress string, senderAddress string, url string, cfg starknet.Config, lggr logger.Logger) (*medianProvider, error) {
-	configProvider, err := NewConfigProvider(chainID, contractAddress, url, cfg, lggr)
+func NewMedianProvider(chainID string, contractAddress string, senderAddress string, basereader starknet.Reader, cfg config.Config, txm txm.TxManager, lggr logger.Logger) (*medianProvider, error) {
+	configProvider, err := NewConfigProvider(chainID, contractAddress, basereader, cfg, lggr)
 	if err != nil {
 		return nil, errors.Wrap(err, "couldn't initialize ConfigProvider")
 	}
 
 	cache := NewTransmissionsCache(cfg, configProvider.reader, configProvider.lggr)
```

Comment:
It's confusing to use the ConfigProvider just as a dependency provider to construct the MedianProvider.

Consider constructing the dependencies inline without using the `NewConfigProvider` here.

Reply:
`errors.Wrap(err, "couldn't initialize ConfigProvider")` is a weird error to get when constructing the `MedianProvider`

Reply:
yeah it's a bit confusing - afaik ConfigProvider is used for bootstrap nodes, and MedianProvider is used for oracle nodes. so the ConfigProvider is still needed for the oracle to func - hence the dependency.

at the very least this could be cleaned up a bit

Reply:
There is no dependency on the `ConfigProvider` per se, we don't use the things it actually provides `ContractConfigTracker` or `OffchainConfigDigester`, but just internal deps the `ConfigProvider` constructs.

I'm suggesting it's cleaner to construct them ourselves in the `NewMedianProvider` or have a separate dependency provider that both can use to construct their dependencies (if the idea is to share code).

Reply:
ohhh interesting...did a little more digging and I understand what you are saying now. the way the caches are set up means we don't actually require most of ConfigProvider (different than solana and terra)

Reply:
wait...on second thought: it does require the config provider dependencies? https://github.com/smartcontractkit/chainlink-relay/blob/24530b163a6f03e71846e669b2c219d7a0f52ede/pkg/types/types.go#L55